--- benchspec/CPU/544.nab_r/src/prm.c	2023-10-02 22:29:26.616416437 +0000
+++ benchspec/CPU/544.nab_r/src/prm.c	2023-10-03 01:08:23.691032936 +0000
@@ -41,7 +41,11 @@
 #include "mpi.h"
 #endif
 
+#ifdef WASM
+extern _Thread_local int errno;
+#else
 extern int errno;
+#endif
 
 static int compressed = 0;
 static char e_msg[256];
@@ -60,7 +64,7 @@
 
 int get_mytaskid(void);
 int rt_errormsg_s(int, char[], char[]);
-void reducerror(int);
+int reducerror(int);
 
 /***********************************************************************
                             GGETS()
